[Merged by Bors] - iOS: replace cargo-lipo, and update for new macOS

mockersf opened the pull request:
# Objective

- Remove `cargo-lipo` as [it's deprecated](https://github.com/TimNN/cargo-lipo#maintenance-status) and doesn't work on new Apple processors
- Fix CI that will fail as soon as GitHub update the worker used by Bevy to macOS 11

## Solution

- Replace `cargo-lipo` with building with the correct target
- Setup the correct path to libraries by using `xcrun --show-sdk-path`
- Also try and fix path to cmake in case it's not found but available through homebrew

mockersf commented on examples/ios/build_rust_deps.sh:
```diff
@@ -1,12 +1,58 @@
+# add path to cmake, needed on apple arm processors as it's not available by default
+if ! cmake --version; then
+    # use the one installed from homebrew
+    if /opt/homebrew/bin/cmake --version; then
```
I'm adding it to path only in case it's installed by homebrew, that's what the `if` is doing hopefully? `bash` is not my preferred language...

HackerFoo replied:
Just add it to the path - it won't cause a problem if the directory doesn't exist.

mockersf commented:
I updated the iOS example to use the new renderer.

I also added a system to move the camera from touch controls, but I'm not sure if how I did touch handling is the best way

cart commented:
Great work!

cart commented:
bors+

cart commented:
bors r+

cart commented:
#3223

cart commented:
bors r+

mockersf commented:
bors crashed with a timeout (https://app.bors.tech/repositories/32050/log)

cart commented:
bors r+

cart commented:
I've seen that once before. Running bors again worked that time.
